src/Rocketlet.ts

```diff
@@ -78,12 +82,18 @@ export abstract class Rocketlet {
      *
      * @return whether the Rocketlet should be enabled or not
      */
-    public abstract onEnable(): boolean;
+    public abstract onEnable(environment: IEnvironmentRead, configurationModify: object): boolean; //todo: Config modify. This should actually be an implementation of configModify which ensures that only own configurations are being modified
```

Comment:
Okay, so to verify I completely understand.

The `initialize()` will be the way that people add things, such as settings, commands, message actions, and etc.

The `onEnable()` and `onDisable()` is where they modify the the items they added, whether to set a setting to enabled/disabled or something else?